pymc/distributions/mixture.py

```diff
@@ -189,7 +195,7 @@ def dist(cls, w, comp_dists, **kwargs):
             # TODO: Allow these to not be a RandomVariable as long as we can call `ndim_supp` on them
             #  and resize them
             if not isinstance(dist, TensorVariable) or not isinstance(
-                dist.owner.op, RandomVariable
+                dist.owner.op, (RandomVariable, SymbolicRandomVariable)
```

Comment:
Thinking out loud: do we get any identifiability problems with nested mixtures? How does posterior sampling perform when a mixture component is a mixture itself? My guess is that things will work out well. Basically, given that a mixture of mixtures is a mixture itself, I'm wondering if there are any computational pitfalls with this approach, i.e. not canonicalizing a graph of nested mixtures.

Perhaps it would a `node_rewriter` of nested mixtures into a single `MarginalMixtureRV` would be an interesting avenue to pursue later...